lib/events/encryptedevent.cpp

```diff
@@ -0,0 +1,41 @@
+#include "encryptedevent.h"
+#include "room.h"
+
+#include <message.h> // QtOlm
+
+using namespace QMatrixClient;
+using namespace QtOlm;
+
+EncryptedEvent::EncryptedEvent(const Room *roomContext, const QJsonObject &ciphertext, const QString &senderKey)
+    : RoomEvent(typeId(), matrixTypeId(),
+                { { AlgorithmKeyL , OlmV1Curve25519AesSha2AlgoKey },
+                  { CiphertextKeyL , ciphertext },
+                  { SenderKeyL, senderKey }
+                })
+{
+    QString identityKey = roomContext->curve25519IdentityKey();
+    QJsonObject personalCipherObject = ciphertext.value(identityKey).toObject();
+    if (personalCipherObject.isEmpty()) {
+        qCDebug(EVENTS) << "Encrypted event is not for the current device";
+        return;
+    }
+    _decryptedEvent = makeEvent<RoomMessageEvent>(roomContext->decryptMessage(personalCipherObject, senderKey.toLatin1()));
+}
+
+EncryptedEvent::EncryptedEvent(const Room *roomContext, QByteArray ciphertext, const QString &senderKey, const QString& deviceId, const QString& sessionId)
+    : RoomEvent(typeId(), matrixTypeId(),
+                { { AlgorithmKeyL , MegolmV1AesSha2AlgoKey },
+                  { CiphertextKeyL , QString(ciphertext) },
+                  { DeviceIdKeyL, deviceId },
+                  { SenderKeyL, senderKey },
+                  { SessionIdKeyL, sessionId },
+                })
+{
+    _decryptedEvent = makeEvent<RoomMessageEvent>(roomContext->decryptMessage(ciphertext, senderKey, deviceId, sessionId));
+}
+
+EncryptedEvent::EncryptedEvent(const QJsonObject &obj)
+    : RoomEvent(typeId(), obj)
+{
+    qCDebug(EVENTS) << "Encrypted event";
```

Comment:
If you're dropping a message to logs, provide at least some identifying information. Nothing's more useless than a row of completely indistinguishable "Encrypted event" lines in logs. At least dumping `id()` wouldn't hurt.